libgraphics/include/pivot/graphics/abstract/AImmediateCommand.hxx

```diff
@@ -36,15 +36,16 @@ public:
 
     template <typename T>
     /// Copy the source buffer into the destination
-    void copyBuffer(AllocatedBuffer<T> &src, AllocatedBuffer<T> &dst)
+    void copyBuffer(AllocatedBuffer<T> &src, AllocatedBuffer<T> &dst, vk::DeviceSize srcOffset = 0,
+                    vk::DeviceSize dstOffset = 0)
     {
         if (src.getBytesSize() > dst.getAllocatedSize())
             throw ImmediateCommandError("The destination buffer is too small");
 
         immediateCommand([&](vk::CommandBuffer &cmd) {
             vk::BufferCopy copyRegion{
-                .srcOffset = 0,
-                .dstOffset = 0,
+                .srcOffset = srcOffset,
+                .dstOffset = dstOffset,
                 .size = src.getBytesSize(),
```

Comment:
Are you sure the size should still be the entire buffer size when an offset is used ?